compiler/src/dotty/tools/dotc/transform/CollectEntryPoints.scala

```diff
@@ -0,0 +1,59 @@
+package dotty.tools.dotc.transform
+
+import dotty.tools.dotc.ast.tpd
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Types
+import dotty.tools.dotc.transform.MegaPhase._
+import java.io.{File => _}
+
+import dotty.tools.dotc.core._
+import SymDenotations._
+import Contexts._
+import Types._
+import Symbols._
+import Phases._
+import dotty.tools.dotc.util.SourcePosition
+import Decorators._
+import StdNames.nme
+import dotty.tools.io.JarArchive
+import dotty.tools.backend.jvm.GenBCode
+
+/**
+ * Small phase to be run to collect main classes and store them in the context.
+ * The general rule to run this phase is either:
+ * - The output of compilation is JarArchive and there is no `-Xmain-class` defined
+ * - The compiler is run from sbt and is forced by flags forcing `ExtractorAPI`
```

Comment:
Looks like this should be removed along with any code related to sbt in this phase since it no longer occurs before ExtractAPI.

Reply:
Fixed